jmclient/jmclient/wallet_rpc.py

```diff
@@ -0,0 +1,684 @@
+                jlog.warn("displaywallet called, but no wallet loaded")
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                print("called displaywallet with wrong wallet")
+                raise InvalidRequestFormat()
+            else:
+                walletinfo = wallet_display(self.wallet_service, False, jsonified=True)
+                return make_jmwalletd_response(request, walletname=walletname, walletinfo=walletinfo)
+
+        @app.route('/session', methods=['GET'])
+        def session(self, request):
+            """ This route functions as a heartbeat, and communicates
+            to the client what the current status of the wallet
+            and services is. TODO: add more data to send to client.
+            """
+            #if no wallet loaded then clear frontend session info
+            #when no wallet status is false
+            session = not self.cookie==None
+            maker_running = False
+            if self.services["maker"] and self.services["maker"].isRunning():
+                maker_running = True
+            if self.taker:
+                coinjoin_in_process = True
+            else:
+                coinjoin_in_process = False
+            if self.wallet_service:
+                if self.wallet_service.isRunning():
+                    wallet_name = self.wallet_name
+                else:
+                    wallet_name = "not yet loaded"
+            else:
+                wallet_name = "None"
+            return make_jmwalletd_response(request,session=session,
+                            maker_running=maker_running,
+                            coinjoin_in_process=coinjoin_in_process,
+                            wallet_name=wallet_name)
+
+        @app.route('/wallet/<string:walletname>/taker/direct-send', methods=['POST'])
+        def directsend(self, request, walletname):
+            """ Use the contents of the POST body to do a direct send from
+            the active wallet at the chosen mixdepth.
+            """
+            self.check_cookie(request)
+            assert isinstance(request.content, BytesIO)
+            payment_info_json = self.get_POST_body(request, ["mixdepth", "amount_sats",
+                                                             "destination"])
+            if not payment_info_json:
+                raise InvalidRequestFormat()
+            if not self.wallet_service:
+                raise NoWalletFound()
+            if not self.wallet_name == walletname:
+                raise InvalidRequestFormat()
+            tx = direct_send(self.wallet_service, int(payment_info_json["amount_sats"]),
+                        int(payment_info_json["mixdepth"]),
+                        destination=payment_info_json["destination"],
+                        return_transaction=True, answeryes=True)
+
+            return make_jmwalletd_response(request,
+                            txinfo=human_readable_transaction(tx))
```

Comment:
Always thinking about machines and not humans, I think we should call `human_readable_transaction(tx, False)`, so that we return a dict instead of a string.
If we do that, we should update the .yaml file too.

Reply:
Done in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/996/commits/71c986b15b624378ff905aba55b3932e798e1ff9